### ISSUE TITLE
registry route addon fails to apply on minishift instance running on the remote machine

### ISSUE THREAD

**Issue by amitkrout** [minishift/minishift#2482]
### General information

  * Minishift version: master
  * OS: Linux (Red Hat Enterprise Linux Server release 7.5 (Maipo))
  * Hypervisor: generic

Docker info on the remote machine
```
# docker version
Client:
 Version:         1.13.1
 API version:     1.26
 Package version: docker-1.13.1-63.git94f4240.el7.x86_64
 Go version:      go1.9.2
 Git commit:      94f4240/1.13.1
 Built:           Mon Apr 30 15:45:42 2018
 OS/Arch:         linux/amd64

Server:
 Version:         1.13.1
 API version:     1.26 (minimum version 1.12)
 Package version: docker-1.13.1-63.git94f4240.el7.x86_64
 Go version:      go1.9.2
 Git commit:      94f4240/1.13.1
 Built:           Mon Apr 30 15:45:42 2018
 OS/Arch:         linux/amd64
 Experimental:    false
```
```
# rpm -qa | grep docker
docker-common-1.13.1-63.git94f4240.el7.x86_64
docker-client-1.13.1-63.git94f4240.el7.x86_64
docker-rhel-push-plugin-1.13.1-63.git94f4240.el7.x86_64
docker-1.13.1-63.git94f4240.el7.x86_64
```

### Steps to reproduce

  1. Start a remote machine (RHEL Server) with all these [configurations](https://github.com/minishift/minishift/blob/master/docs/source/using/experimental-features.adoc#run-against-an-existing-machine)
  2. start minishift from the host machine
```
$ export MINISHIFT_ENABLE_EXPERIMENTAL=y
$ ./minishift start --vm-driver generic --remote-ipaddress <ip address of the remote machine> --remote-ssh-user root --remote-ssh-key ~/.ssh/id_rsa
```
  3. After the successful minishift start, apply registry-route addon
```
$ ./minishift addons apply registry-route
-- Applying addon 'registry-route':....Error applying the add-on: Error executing command 'ssh command error:
command : sudo chown docker /etc/docker/certs.d/docker-registry-default.10.70.45.10.nip.io
err     : exit status 1
output  : chown: invalid user: ‘docker’
':
```
  4. 

### Expected
registry-route addon should be applied successfully

### Actual
Failing

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by praveenkumar** [comment]
This I think should be very hard to fix, we need to add username also as part of addon context, will need to take close look to it.

**Comment by LalatenduMohanty** [comment]
Resolved via #2487 https://github.com/minishift/minishift/commit/5999bea2195ac8ac1581be6b39e0c029243549d1
